### PR TITLE
add a test summary that we can mark as a blocker

### PR DESCRIPTION
**Pull request by joejulian**
This adds a "summary" job that will pass or fail depending on the pass/fail state of all the matrix tests. The summary test can then be added to the required status checks.

### CODE REVIEW

File: .github/workflows/pull_requests.yaml [hunk 1]
```diff
@@ -162,3 +162,13 @@ jobs:
             --skip-missing-values \
             --excluded-charts console \
             --target-branch ${{ github.event.repository.default_branch }}
+  summary:
+    if: always()
+    needs:
+      - test
```
Comment (RafalKorepta) [inline .github/workflows/pull_requests.yaml:168]:
```suggestion
    needs:
      - test
      - lint
```

File: .github/workflows/pull_requests.yaml [hunk 2]
```diff
@@ -162,3 +162,13 @@ jobs:
...
+    runs-on: ubuntu-latest
```
Comment (RafalKorepta) [inline .github/workflows/pull_requests.yaml:169]:
```suggestion
    runs-on: ubuntu-22.04
```

File: .github/workflows/pull_requests.yaml [hunk 3]
```diff
@@ -162,3 +162,13 @@ jobs:
...
+    steps:
+      - name: Summarize the results of the test matrix pass/fail
+        uses: re-actors/alls-green@release/v1
+        with:
+          jobs: ${{ toJSON(needs)}}
```
Comment (RafalKorepta) [inline .github/workflows/pull_requests.yaml:174]:
```suggestion
        with:
          allowed-failures: check-values
          jobs: ${{ toJSON(needs)}}
```

File: .github/workflows/pull_requests.yaml (hunk 1 above)
Comment (joejulian) [reply .github/workflows/pull_requests.yaml:168]: No, those can be checked directly. Only the matrix tests need this.

File: .github/workflows/pull_requests.yaml (hunk 3 above)
Comment (joejulian) [reply .github/workflows/pull_requests.yaml:174]: again, this is only for the matrix tests.

**Review by RafalKorepta — APPROVED** [review]

**Pull request merged**